src/main/java/mate/academy/AsyncRequestProcessor.java

```diff
@@ -11,6 +13,19 @@ public AsyncRequestProcessor(Executor executor) {
     }
 
     public CompletableFuture<UserData> processRequest(String userId) {
-        return null;
+        Map<String, UserData> userDataMap = new ConcurrentHashMap<>();
+
+        if (userDataMap.containsKey(userId)) {
+            try {
+                Thread.sleep(500);
+            } catch (InterruptedException e) {
+                throw new RuntimeException();
```

Comment:
Catching `InterruptedException` and throwing a new `RuntimeException` without any message or context is not ideal. It is better to either restore the interrupt status with `Thread.currentThread().interrupt()` or throw a custom exception with more context about the error.